### ISSUE TITLE
Error importing from keras h5 - BatchNormalizationV1

### ISSUE THREAD

**Issue by talvasconcelos** [tensorflow/tfjs#1334]
TensorFlow.js version: 1.0.0

I'm getting thie bellow error when importing a keras model from Colab.
tensorflow version in colab is 1.13.1

I found issue #1255 on this matter but i couldn't get it to work!

Error: Unknown layer: BatchNormalizationV1. This may be due to one of the following reasons:
1. The layer is defined in Python, in which case it needs to be ported to TensorFlow.js or your JavaScript code.
2. The custom layer is defined in JavaScript, but is not registered properly with tf.serialization.registerClass().
    at new t (tf-layers.esm.js?28b0:17)
    at deserializeKerasObject (tf-layers.esm.js?28b0:17)
    at deserialize (tf-layers.esm.js?28b0:17)
    at t.fromConfig (tf-layers.esm.js?28b0:17)
    at deserializeKerasObject (tf-layers.esm.js?28b0:17)
    at deserialize (tf-layers.esm.js?28b0:17)
    at eval (tf-layers.esm.js?28b0:17)
    at eval (tf-layers.esm.js?28b0:17)
    at Object.eval [as next] (tf-layers.esm.js?28b0:17)
    at o (tf-layers.esm.js?28b0:17)

**Comment by davidsoergel** [comment]
It's a known remaining bug in the tf.Keras exporter that this happens when saving to HDF5 (.h5) with `model.save()`, even though it was fixed in `model.to_json()`.  The additional fix will likely be included in a TF 2.0.x alpha release in the coming weeks.

In the meantime, the most expedient fix is to manually edit your converted model.json to replace instances of `BatchNormalizationV1` with `BatchNormalization`.

**Comment by caisq** [comment]
@talvasconcelos Sorry about the inconvenience. See @davidsoergel's comment above. @fchollet and @karmel and @YHLIANG on the tf.keras team are aware of this issue are actively working towards a fix.

The current workaround is to use "vanilla" keras (i.e., `import keras`) to save your model, assuming you are not using any tf.keras-specific features.

**Comment by caisq** [comment]
@talvasconcelos Also, editing the content of model.json (after the python-to-javascript conversion step) is another possible workaround, as @davidsoergel pointed out.

**Comment by talvasconcelos** [comment]
Hello all, thanks for your awnsers. I tried editing the final json, but i must have screwed something up. Anyway, i did a workaround... i trained in Colab TPU, with tf.keras, then saved the wheigths. Remade rhe model in vanilla keras and imported the weights. It seems to be working, still need to test in js.

**Comment by RadEdje** [comment]
> It's a known remaining bug in the tf.Keras exporter that this happens when saving to HDF5 (.h5) with `model.save()`, even though it was fixed in `model.to_json()`. The additional fix will likely be included in a TF 2.0.x alpha release in the coming weeks.
> 
> In the meantime, the most expedient fix is to manually edit your converted model.json to replace instances of `BatchNormalizationV1` with `BatchNormalization`.

Hi. I tried this:
```
edit your converted model.json to replace instances of `BatchNormalizationV1` with `BatchNormalization`.
```
It's running in the browser. but it's taking forever to load just 3 shard.bin files... why is there a bin extension. the old shards (using older converters) did not have a bin extension...

**Comment by RadEdje** [comment]
> @talvasconcelos Sorry about the inconvenience. See @davidsoergel's comment above. @fchollet and @karmel and @YHLIANG on the tf.keras team are aware of this issue are actively working towards a fix.
> 
> The current workaround is to use "vanilla" keras (i.e., `import keras`) to save your model, assuming you are not using any tf.keras-specific features.

how can I do this?
I've used if used tf.keras.appilcations for mobileNet_V2.... and saved the model as simlply "model"

I assume that when I use

```
mode.save
```

here I'm using the tf.keras api.

how do i get this tf.keras model as a vanilla keras model and then save that using the vanilla keras model.save api?

thanks for any suggestions.

**Comment by pankajksingh** [comment]
I am using tf.keras api to build sequence to sequence encoder-decoder model.
I tried the following:
1. Save and load full model using save() & load_model() functions.
2. Save model & weights separately. For saving model I used to_json() function. For laoding it back, I am using model_from_json().

In either case, I am encountering the error
**ValueError: Unknown layer: BatchNormalizationV1**

When can we expect a fix for this?

Sharing version of libraries:
keras version : 2.2.4
tensorflow version :  **1.13.1** 

I will try to modify the model json from BatchNormalizationV1 to BatchNormalization and see if that helps.
